### PR TITLE
enh(plugin)add ec2 spot fleet discovery

### PR DESCRIPTION
**Pull request by Sims24**
- add discovery mode to get spot fleet requests

### CODE REVIEW

**Comment by garnier-quentin** [comment]
Could you add paws also ?

File: cloud/aws/ec2/mode/discoveryspotfleetrequests.pm [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+    $self->{output}->exit();
+}
+
+1;
+
+__END__
+
+=head1 MODE
+
+EC2 Sport Fleet Requests discovery.
```
Comment (thibaults-centreon) [inline cloud/aws/ec2/mode/discoveryspotfleetrequests.pm:101]:
```suggestion
EC2 Spot Fleet Requests discovery.

```

File: cloud/aws/ec2/mode/discoveryspotfleetrequests.pm [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    my $disco_stats;
+
+    $disco_stats->{start_time} = time();
+
+    my %asgs;
+
+    my $spot_fleet_requests = $options{custom}->discovery(
+        region => $self->{option_results}->{region},
+        service => 'ec2',
+        command => 'describe-spot-fleet-requests'
```
Comment (thibaults-centreon) [inline cloud/aws/ec2/mode/discoveryspotfleetrequests.pm:60]:
```suggestion
    my $spot_fleet_requests = $options{custom}->discovery(
        region  => $self->{option_results}->{region},
        service => 'ec2',
        command => 'describe-spot-fleet-requests'

```

File: cloud/aws/ec2/mode/discoveryspotfleetrequests.pm [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+use JSON::XS;
+
+sub new {
+    my ($class, %options) = @_;
+    my $self = $class->SUPER::new(package => __PACKAGE__, %options);
+    bless $self, $class;
+    
+    $options{options}->add_options(arguments => {
+        "prettify"      => { name => 'prettify' },
+        "filter-type:s" => { name => 'filter_type' },
```
Comment (thibaults-centreon) [inline cloud/aws/ec2/mode/discoveryspotfleetrequests.pm:36]: Shouldn't it be "filter-state" ?

File: cloud/aws/custom/awscli.pm [hunk 4]
```diff
@@ -399,6 +399,35 @@ sub ec2spot_get_active_instances_status {
...
+
+    my $cmd_options = $self->ec2spot_list_fleet_requests_set_cmd(%options);
+    my $raw_results = $self->execute(cmd_options => $cmd_options);
+
+    my $resource_results = [];
+    foreach my $fleet_request (@{$raw_results->{SpotFleetRequestConfigs}}) {
+        push @{$resource_results}, {
+            SpotFleetRequestState => $fleet_request->{SpotFleetRequestState},
+            SpotFleetRequestId => $fleet_request->{SpotFleetRequestId},
+            ActivityStatus => $fleet_request->{ActivityStatus},
```
Comment (thibaults-centreon) [inline cloud/aws/custom/awscli.pm:424]:
```suggestion
            SpotFleetRequestState => $fleet_request->{SpotFleetRequestState},
            SpotFleetRequestId    => $fleet_request->{SpotFleetRequestId},
            ActivityStatus        => $fleet_request->{ActivityStatus}
```

File: cloud/aws/ec2/mode/discoveryspotfleetrequests.pm (hunk 3 above)
Comment (Sims24) [reply cloud/aws/ec2/mode/discoveryspotfleetrequests.pm:36]: definitely !

**Pull request merged**